src/cli_plugin/install/cleanup.js

```diff
@@ -0,0 +1,32 @@
+import rimraf from 'rimraf';
+import fs from 'fs';
+
+export function cleanPrevious(settings, logger) {
+  return new Promise(function (resolve, reject) {
+    try {
+      fs.statSync(settings.workingPath);
+
+      logger.log('Found previous install attempt. Deleting...');
+      try {
+        rimraf.sync(settings.workingPath);
+      } catch (e) {
+        reject(e);
+      }
+      resolve();
+    } catch (e) {
+      if (e.code !== 'ENOENT') reject(e);
+
+      resolve();
+    }
+  });
+};
+
+export function cleanError(settings) {
+  // delete the working directory.
+  // At this point we're bailing, so swallow any errors on delete.
+  try {
+    rimraf.sync(settings.workingPath);
+    rimraf.sync(settings.plugins[0].path);
```

Comment:
I know the intention of the plugin installer right now is to ignore all but the first plugin, but perhaps we could encapsulate that logic in a single place?  In that case, code like this becomes more expressive, and we don't litter the implementation detail of grabbing the first plugin in the array throughout the rest of the module.